src/components/manifold-subscription-details/components/PlanSelector.tsx

```diff
@@ -7,9 +7,47 @@ import {
   PlanFixedFeatureEdge,
   PlanMeteredFeatureEdge,
   PlanConfigurableFeatureEdge,
+  UpdateSubscriptionMutation,
+  UpdateSubscriptionMutationVariables,
 } from '../../../types/graphql';
 import state from '../store';
 
+import updateSubscriptionMutation from '../update-subscription.graphql';
+import { toFeatureMap } from '../../../utils/plan';
+
+const updateSubscription = async () => {
+  if (!state.connection || !state.planId) {
+    return;
+  }
+
+  state.isUpdating = true;
+
+  const variables: UpdateSubscriptionMutationVariables = {
+    id: state.subscriptionId,
+    planId: state.planId,
+    configuredFeatures: Object.entries(state.configuredFeatures).map(([label, value]) => ({
+      label,
+      value: `${value}`,
+    })),
+  };
+
+  const { data, errors } = await state.connection.graphqlFetch<UpdateSubscriptionMutation>({
+    query: updateSubscriptionMutation,
+    variables,
+  });
+
+  state.isUpdating = false;
+  state.isEditing = false;
+  const id = data?.updateSubscription?.data?.plan?.id;
```

Comment:
Not related to this PR, but why does our GraphQL API namespace like this? Isn’t `data.plan` more predictable?

Reply:
Not sure that's something we can decide since this is the result of the `updateSubscription` mutation. if there were multiple mutations we need this namespacing to separate the results. 

Perhaps we could get away with not having the second `data` though? I'm not sure if errors would ever appear at that level. I assume they only exist at the top level, so having a second `data` field that's nested like that might be unnecessary.

Reply:
Ah that makes sense. I didn’t think about multiple mutations.